internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala

```diff
@@ -162,13 +162,13 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
         case _ => newOne()
       }
     }
-    private def addClassDependency(deps: HashSet[ClassDependency], fromClass: Symbol, dep: Symbol): Unit = {
+    private def addClassDependency(deps: HashSet[ClassDependency], fromClass: Symbol, dep: Symbol): Unit = if (dep != NoSymbol) {
```

Comment:
Perfect, even better. But I think we should guard against all of this in the whole Zinc. I'll do that in a PR.